src/test/compile-fail/explicit-self-lifetime-mismatch.rs

```diff
@@ -1,3 +1,5 @@
+//~  ERROR mismatched types
```

Comment:
This doesn't make sense - how are you getting mismatched type errors on this line?

Reply:
Ya so this is the thing I was having trouble with on Gitter. In https://github.com/rust-lang/rust/pull/45870/commits/e06cd316a49556bc80e8284c2dab6ac094f75c5f#diff-15d40489387405cc3e23e2f9f66a74c2L519 I removed some error checks that I (still think) are redundant, and it caused the error messages to go wonky. I can't figure out why